torchvision/prototype/transforms/functional/_color.py

```diff
@@ -98,7 +98,98 @@ def adjust_sharpness(inpt: features.InputTypeJIT, sharpness_factor: float) -> fe
         return adjust_sharpness_image_pil(inpt, sharpness_factor=sharpness_factor)
 
 
-adjust_hue_image_tensor = _FT.adjust_hue
+def _rgb_to_hsv(image: torch.Tensor) -> torch.Tensor:
+    r, g, _ = image.unbind(dim=-3)
+
+    # Implementation is based on https://github.com/python-pillow/Pillow/blob/4174d4267616897df3746d315d5a2d0f82c656ee/
+    # src/libImaging/Convert.c#L330
+    minc, maxc = torch.aminmax(image, dim=-3)
+
+    # The algorithm erases S and H channel where `maxc = minc`. This avoids NaN
+    # from happening in the results, because
+    #   + S channel has division by `maxc`, which is zero only if `maxc = minc`
+    #   + H channel has division by `(maxc - minc)`.
+    #
+    # Instead of overwriting NaN afterwards, we just prevent it from occuring so
+    # we don't need to deal with it in case we save the NaN in a buffer in
+    # backprop, if it is ever supported, but it doesn't hurt to do so.
+    eqc = maxc == minc
+
+    cr = maxc - minc
```

Comment:
Could you make these variable names more expressive? It is hard to understand what is actually happening if the values are hidden behind acronyms. I'm totally ok with `r`, `g`, and `b` and in turn with `h`, `s`, and `v`. But `eqc`, `cr`, `rc`, ... are opaque. I'm aware that PIL uses the same names, but I would prefer to have expressive names in our implementation and maybe a comment stating the corresponding PIL variable.

Reply:
The point of this PR is to optimize the runtime reusing existing code. Some vars are just temporary vars. Others have a certain meaning (to me): cr = channel_range. I'll replace cr with `channel_range`, for others I do not have a good name.

Reply:
`cr` to `channel_range` is definitely an improvement!